src/Bearded.TD/Game/Simulation/Elements/ElementalPhenomenonScopeBase.cs

```diff
@@ -28,57 +32,48 @@ public void Adopt(TEffect effect, Instant now)
 
     public void ApplyTick(Instant now)
     {
-        activeEffects.RemoveAll(e => e.Expiry <= now);
+        activeEffects.RemoveAll(now, static (t, e) => e.Expiry <= t);
 
-        if (TryChooseEffect(out var effect))
+        if (TryChooseEffect(CollectionsMarshal.AsSpan(activeEffects)) is { } effect)
```

Comment:
We're using a complex span mechanism to avoid allocating an `IEnumerator` (I think?), yet we now allocate a boxed `TEffect` for the return. Are you sure this is a net improvement? What am I missing?